proxy.go

```diff
@@ -188,6 +216,207 @@ func (proxy *proxy) releaseToken(token Token) (*tokenInfo, error) {
 	return info, nil
 }
 
+var storeStateDir = "/var/lib/clear-containers/proxy/"
+
+type proxyStateOnDisk struct {
+	SocketPath      string   `json:"socket_path"`
+	EnableVMConsole bool     `json:"enable_vm_console"`
+	ContainerIDs    []string `json:"container_ids"`
+}
+
+// returns false if it's a clean start (i.e. no state is stored) or restoring failed
+func (proxy *proxy) restoreState() bool {
+	proxyStateFilePath := storeStateDir + "proxy_state.json"
+	if _, err := os.Stat(storeStateDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(storeStateDir, 0755); err != nil {
+			proxyLog.Errorf("Couldn't create directory %s: %v",
+				storeStateDir, err)
+		}
+		return false
+	}
+	if _, err := os.Stat(proxyStateFilePath); os.IsNotExist(err) {
```

Comment:
Again, this isn't required - just call `ioutil.ReadFile()` and have it fail if the file doesn't exist.

Reply:
Agree, thanks!